packages/docusaurus-theme-common/src/utils/codeBlockUtils.ts

```diff
@@ -148,6 +153,14 @@ export function parseLines(
           highlightRange += `${highlightBlockStart!}-${lineNumber - 1},`;
           break;
 
+        case 'sample-start':
```

Comment:
wonder if sample is the best marker name? 🤔

Reply:
This is directly from the Kotlin docs example in #2215: `// sampleStart` in that case. We can certainly bikeshed about this, and if we are to just build a generic API, maybe even no bikeshedding at all :D